feat(user-profile): improve user profile header to match with design

arathjz opened the pull request:
**What:**
Update user profile header to match with the design

**Why:**
To address: https://app.asana.com/0/1159164196409129/1186341795801831/f

**How:**
- Display user profile always expanded
- Hide button to collapse/expand profile
- Update styles on a user profile header

Media:
- [Admin](https://recordit.co/i2pTjw2BEB)
- [Normal user](https://recordit.co/BBTEiJzYTh)

orlando commented on javascripts/discourse/widgets/dc-topic-title.js.es6:
```diff
@@ -27,15 +27,17 @@ export default createWidget("dc-topic-title", {
 
   buildTopicInfo(attrs) {
-    return [this.buildCategory(attrs), this.buildTopicTags(attrs)];
+    return [this.buildCategory(attrs), this.buildTopicTags(attrs)].filter(Boolean);
```
@arathjz what's the idea behind this?

arathjz replied:
Now that I think about it does it is not needed at all. It was just in case any of the two functions returned null or undefined

orlando commented on scss/templates/user.scss:
```diff
@@ -93,3 +93,103 @@
+    }
+  }
+  // This is the only way to hide the Collapse/Expand button
+  [href=""] {
```
@arathjz what element are we targeting here? I guess is `a` but if that's the case why we target the `href`?

arathjz replied:
That's due for some reason, the button collapse it's an anchor with an empty href, it does not have either id or name attribute to target to. These fellas made weird sorcery while building some parts of the app

```html
<a href="" class="btn btn-default" data-ember-action="" data-ember-action-1147="1147">
  <svg class="fa d-icon d-icon-angle-double-up svg-icon svg-string" 
  xmlns="http://www.w3.org/2000/svg"><use xlink:href="#angle-double-up"></use></svg>  
  Collapse
</a>
```

![image](https://user-images.githubusercontent.com/20747535/89240668-7023fd80-d5c2-11ea-852b-2ea65d77536a.png)

arathjz commented on javascripts/discourse/widgets/dc-topic-title.js.es6:
```diff
@@ -27,15 +27,17 @@ export default createWidget("dc-topic-title", {
-        href: `/tag/${tag}`
-      }
-    }, tag));
+    if (tags) {
```
Agree, there are many ways to approach this, I'll go by your suggestion, thanks man

arathjz commented on scss/templates/user.scss:
```diff
@@ -93,3 +93,103 @@
+    align-items: center;
+    background-color: transparent;
+    border-radius: 0.25rem;
+    border: 1px solid $blue;
```
My bad dude haha I always forget to use rem in borders

arathjz commented:
@duranmla I've addressed your comments. From now on I'll add my style rules focused on mobile-first, and I'll be careful when adding borders (I always forget to use rems on borders haha). Thanks for the feedback man! 🚀

arathjz commented:
@duranmla Let's address the separation of the buttons on another PR
